apps/web/src/pages/ApplicantDashboardPage/components/MetadataDate.tsx

```diff
@@ -0,0 +1,152 @@
+
+  const isDraftStatus = status === ApplicationStatus.DRAFT;
+  const isExpiredStatus = status === ApplicationStatus.EXPIRED;
+  if (isDraftStatus || isExpiredStatus) {
+    return (
+      <span>
+        {intl.formatMessage({
+          defaultMessage: "Deadline",
+          id: "nIAA4Q",
+          description: "Label for deadline metadata",
+        })}{" "}
+        {intl.formatMessage(commonMessages.dividingColon)}{" "}
+        {closingDate
+          ? formatDate({
+              date: parseDateTimeUtc(closingDate),
+              formatString: "PPP",
+              intl,
+              timeZone: "Canada/Pacific",
+            })
+          : nullMessage}
+      </span>
+    );
+  }
+
+  const isReceivedStatus = status === ApplicationStatus.RECEIVED;
+  const isUnderReviewStatus = status === ApplicationStatus.UNDER_REVIEW;
+  const isUnderAssessmentStatus = status === ApplicationStatus.UNDER_ASSESSMENT;
+  if (isReceivedStatus || isUnderReviewStatus || isUnderAssessmentStatus) {
+    return (
+      <span>
+        {intl.formatMessage({
+          defaultMessage: "Submitted",
+          id: "B1sXVl",
+          description: "Label for submitted metadata",
+        })}{" "}
+        {intl.formatMessage(commonMessages.dividingColon)}{" "}
+        {submittedAt
+          ? formatDate({
+              date: parseDateTimeUtc(submittedAt),
+              formatString: "PPP",
+              intl,
+              timeZone: "Canada/Pacific",
+            })
+          : nullMessage}
+      </span>
+    );
+  }
+
+  const isSuccessfulStatus = status === ApplicationStatus.SUCCESSFUL;
+  const isUnsuccessfulStatus = status === ApplicationStatus.UNSUCCESSFUL;
+  if (isSuccessfulStatus || isUnsuccessfulStatus) {
+    return (
+      <span>
+        {intl.formatMessage({
+          defaultMessage: "Assessed",
+          id: "GYcxDu",
+          description: "Label for assessed metadata",
+        })}{" "}
+        {intl.formatMessage(commonMessages.dividingColon)}{" "}
+        {finalDecisionAt
+          ? formatDate({
+              date: parseDateTimeUtc(finalDecisionAt),
+              formatString: "PPP",
+              intl,
+              timeZone: "Canada/Pacific",
+            })
+          : nullMessage}
+      </span>
+    );
+  }
+
+  return null;
+};
+
+export const RecruitmentDate = ({
+  finalDecisionAt,
+  removedAt,
+  status,
+}: {
+  finalDecisionAt?: string | null;
+  removedAt?: string | null;
+  status: ApplicationStatus;
+}) => {
+  const intl = useIntl();
+  const nullMessage = intl.formatMessage(commonMessages.notFound);
+
+  const isOpenToJobsStatus = status === ApplicationStatus.OPEN_TO_JOBS;
+  const isNotInterestedStatus = status === ApplicationStatus.NOT_INTERESTED;
+  const isHiredStatus = status === ApplicationStatus.HIRED;
+  const isExpiredStatus = status === ApplicationStatus.EXPIRED;
+  if (
+    isOpenToJobsStatus ||
+    isNotInterestedStatus ||
+    isHiredStatus ||
+    isExpiredStatus
+  ) {
```

Comment:
We seem to be doing this pattern a lot where we assign status checks to a variable and check it immediatley.

I wonder if we could create some arrays that group status and do a single `statusArr.includes(status)` instead :thinking:

Reply:
We have to get this in asap since it's blocking a lot of other issues, so I'll make a ticket to refactor all places using this pattern and change it to the way you mentioned. That work?

Reply:
@mnigh Is something confusing?

Reply:
> @mnigh Is something confusing?

If it will take a long time to do this refactor, sure, push it to another issue but if @esizer has a solution already, why not do it right the first time instead of pushing a slower solution?